docs/knowledgebase/advanced/consensus.md

```diff
@@ -4,14 +4,17 @@ title: Consensus
 
 Blockchain nodes use consensus engines to agree on the blockchain's state. This article covers the
 fundamentals of consensus in blockchain systems, how consensus interacts with the runtime in the
-Substrate framework, and the consensus engines available with the framework.
+Substrate framework, and the consensus engines available within Substrate.
 
 ## State Machines and Conflicts
 
 A blockchain runtime is a [state machine](https://en.wikipedia.org/wiki/Finite-state_machine). It
 has some internal state, and state transition function that allows it to transition from its current
 state to a future state. In most runtimes there are states that have valid transitions to multiple
-future states, but a single transition must be selected.
+future states, but a single transition must be selected. In some state machine models, it is also possible to associate actions with a state:
+
+- an entry action: performed when entering the state, and
+- an exit action: performed when exiting the state.
```

Comment:
I'm personally interested in this because I think a lot about state machines. Have a link I could read?

When I originally wrote this article I had a whole section about conflict exclusion before getting to forks, but we agreed to remove it because it was not relevant to Substrate which considers any fork a conflict even if the blocks could commute. I feel the same about this comment regarding alternate state machines.

Reply:
What does this addition mean? Entry and exit actions have nothing to do with the state.